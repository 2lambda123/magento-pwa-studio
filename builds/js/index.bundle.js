window.addEventListener("load",e=>{if(document.getElementsByClassName("table-of-contents")){let e=document.getElementsByClassName("active")[0];e&&e.scrollIntoView({inline:"center",block:"center",behavior:"smooth"})}});