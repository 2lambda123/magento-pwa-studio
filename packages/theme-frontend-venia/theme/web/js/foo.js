console.log('logging from rush/web');
