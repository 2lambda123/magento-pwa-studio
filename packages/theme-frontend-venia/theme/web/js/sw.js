/**
 * Welcome to your Workbox-powered service worker!
 *
 * You'll need to register this file in your web app and you should
 * disable HTTP caching for this file too.
 * See https://goo.gl/nhQhGp
 *
 * The rest of the code is auto-generated. Please don't update this file
 * directly; instead, make changes to your Workbox build configuration
 * and re-run your build process.
 * See https://goo.gl/2aRDsh
 */

importScripts("https://storage.googleapis.com/workbox-cdn/releases/3.0.0-beta.2/workbox-sw.js");

importScripts(
  "https://magento-venia1.local.pwadev:8001/pub/static/frontend/Magento/venia/en_US/precache-manifest.8cabad7244ec78b69a84d76533201b8a.js"
);

workbox.skipWaiting();

/**
 * The workboxSW.precacheAndRoute() method efficiently caches and responds to
 * requests for URLs in the manifest.
 * See https://goo.gl/S9QRab
 */
self.__precacheManifest = [].concat(self.__precacheManifest || []);
workbox.precaching.suppressWarnings();
workbox.precaching.precacheAndRoute(self.__precacheManifest, {});

workbox.routing.registerRoute(/https:\/\/magento-ux.github.io\/pwaza/, workbox.strategies.staleWhileRevalidate(), 'GET');
