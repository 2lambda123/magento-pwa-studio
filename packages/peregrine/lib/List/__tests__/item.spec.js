import React from 'react';

import { Item } from '../index.js';
import createTestInstance from '../../util/createTestInstance';

const item = {
    id: '001',
    name: 'Test Product 1',
    small_image: '/test/product/1.png',
    price: {
        regularPrice: {
            amount: {
                value: 100
            }
        }
    }
};

const classes = {
    root: 'abc'
};

const hasFocus = jest.fn();

const render = jest.fn(() => <div />);

const updateSelection = jest.fn();

const setFocus = jest.fn();

test('snapshot testing Item', () => {
    const renderer = createTestInstance(
        <Item
            uniqueID="001"
            classes={classes}
            hasFocus={hasFocus}
            isSelected={false}
            item={item}
            itemIndex={0}
            render={render}
            updateSelection={updateSelection}
            setFocus={setFocus}
        />
    );

    expect(renderer.toTree()).toMatchSnapshot();
});

test('onClick function is memoized on updateSelection prop', () => {
    const renderer = createTestInstance(
        <Item item={item} updateSelection={updateSelection} />
    );
    const firstOnClick = renderer.root.children[0].props.onClick;
    renderer.update(<Item item={item} updateSelection={updateSelection} />);

    expect(firstOnClick).toEqual(renderer.root.children[0].props.onClick);

    renderer.update(<Item item={item} updateSelection={jest.fn()} />);

    expect(firstOnClick).not.toEqual(renderer.root.children[0].props.onClick);
});

test('onFocus function is memoized on setFocus prop', () => {
    const renderer = createTestInstance(
        <Item item={item} setFocus={setFocus} />
    );
    const firstOnFocus = renderer.root.children[0].props.onFocus;
    renderer.update(<Item item={item} setFocus={setFocus} />);

    expect(firstOnFocus).toEqual(renderer.root.children[0].props.onFocus);

    renderer.update(<Item item={item} setFocus={jest.fn()} />);

    expect(firstOnFocus).not.toEqual(renderer.root.children[0].props.onFocus);
});
