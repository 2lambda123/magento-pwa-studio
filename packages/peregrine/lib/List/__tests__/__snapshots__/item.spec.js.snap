// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing Item 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "classes": Object {
      "root": "abc",
    },
    "hasFocus": [MockFunction],
    "isSelected": false,
    "item": Object {
      "id": "001",
      "name": "Test Product 1",
      "price": Object {
        "regularPrice": Object {
          "amount": Object {
            "value": 100,
          },
        },
      },
      "small_image": "/test/product/1.png",
    },
    "itemIndex": 0,
    "render": [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "children": null,
            "className": "abc",
            "classes": Object {
              "root": "abc",
            },
            "hasFocus": [MockFunction],
            "isSelected": false,
            "item": Object {
              "id": "001",
              "name": "Test Product 1",
              "price": Object {
                "regularPrice": Object {
                  "amount": Object {
                    "value": 100,
                  },
                },
              },
              "small_image": "/test/product/1.png",
            },
            "itemIndex": 0,
            "onClick": [Function],
            "onFocus": [Function],
          },
          Object {},
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": <div />,
        },
      ],
    },
    "setFocus": [MockFunction],
    "uniqueID": "001",
    "updateSelection": [MockFunction],
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "children": null,
      "className": "abc",
      "classes": Object {
        "root": "abc",
      },
      "hasFocus": [MockFunction],
      "isSelected": false,
      "item": Object {
        "id": "001",
        "name": "Test Product 1",
        "price": Object {
          "regularPrice": Object {
            "amount": Object {
              "value": 100,
            },
          },
        },
        "small_image": "/test/product/1.png",
      },
      "itemIndex": 0,
      "onClick": [Function],
      "onFocus": [Function],
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "host",
      "props": Object {},
      "rendered": Array [],
      "type": "div",
    },
    "type": [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "children": null,
            "className": "abc",
            "classes": Object {
              "root": "abc",
            },
            "hasFocus": [MockFunction],
            "isSelected": false,
            "item": Object {
              "id": "001",
              "name": "Test Product 1",
              "price": Object {
                "regularPrice": Object {
                  "amount": Object {
                    "value": 100,
                  },
                },
              },
              "small_image": "/test/product/1.png",
            },
            "itemIndex": 0,
            "onClick": [Function],
            "onFocus": [Function],
          },
          Object {},
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": <div />,
        },
      ],
    },
  },
  "type": [Function],
}
`;
